### DISCUSSION TITLE
Unable to use all cores with mpiexec

### DISCUSSION THREAD

**Discussion by msimmar26** [firemodels/fds#12802] [Q&A]
Hi,
I'm attempting to execute Parallel-Computer processing with _mpiexec_ but the _mpi processes_ get stuck. I wonder if there is a limit of computers or CPUs per computers to be used, or a special configuration needs to be done in my computers network.
Both mpi_test and a simple simulation work when I invoke just 1 process per computer in my hosts.txt file.

Hosts file for 3 MPI processes (this case works):
computerA : 1
computerB : 1
computerC : 1

But when I set more than 1 MPI process per computer the process shows "Starting FDS..." but never carries on (this case doesn't work):
computerA : 1
computerB : 2

Here are the commands I'm using:
>mpiexec -register
User: xxxx
password: xxxx

>set I_MPI_PORT_RANGE=54912:54912  #Port 54912 has been opened previously to avoid Firewall Action

>mpiexec -wdir \Marina\Test_CFG2\Prova_1 -machine \Marina\Test_CFG2\Prova_1\hosts.txt test_mpi

>mpiexec -wdir \Marina\Test_CFG2\Prova_1 -machine \Marina\Test_CFG2\Prova_1\hosts.txt fds \Marina\Test_CFG2\Prova_1\room3.fds

I'm working with a set of 20 computers all of them connected within the same Windows network (my University network). The simulations are run in the same shared working directory. I've also tried the same simulations with a private network of only 3 computers (out of the University network), without firewall, but the same problem occurs, so it is not a problem of firewall nor the university network.

Here are the specifications of the computers I'm using (in case the problem is derived from the PCs):

              Intel(R) Core(TM) i5-6400 CPU @ 2.70GHz
              Base speed: 2.70 GHz
              Sockets: 1
              Cores: 4
              Logical processors: 4
              Virtualization: Enabled
              L1 cache: 256 kB
              L2 cache: 1.0 MB
              L3 cache: 6.0 MB

And this is the script for the simple simulation:

&HEAD CHID='room3' TITLE='One room garden house example internet' /

&MESH IJK=15,30,20 XB=0,1.5,0,3,0,2, MPI_PROCESS=0 / 3m x 3m x 2m
&MESH IJK=15,30,20 XB=1.5,3,0,3,0,2, MPI_PROCESS=1 / 3m x 3m x 2m
&MESH IJK=30,30,20 XB=0,3,0,3,2,4, MPI_PROCESS=2 / 3m x 3m x 2m

&MISC SHARED_FILE_SYSTEM = .TRUE. /

&TIME T_END=30./

&DUMP NFRAMES=30/

&VENT XB= 0,0,0.5,2.5,0,1.5, SURF_ID='OPEN'/ PORTA
&VENT XB= 0.5,1.5,0,0,1,1.5, SURF_ID='OPEN'/ FINESTRA
&VENT XB= 0.5,1.5,3,3,1,1.5, SURF_ID='OPEN'/ FINESTRA

&SURF ID='fire', HRRPUA=125 /
&OBST XB= 2.5,3,2.5,3,0,0.2 SURF_ID='fire' /

&REAC FUEL='ACETONE' /

&SLCF PBY=2.75, QUANTITY='TEMPERATURE' /

&SLCF PBX=2.75, QUANTITY='TEMPERATURE'

&DEVC XBP=2.75,2.75,2.75,2.75,0.2,2, QUANTITY='TEMPERATURE', ID='THERMOCOUPLES', POINTS=9, POINTS_ARRAY_Z=0.2,0.4,0.6,0.8,1,1.2,1.4,1.6,1.8 /

&TAIL/

So, what do you think the problem could be? I've tried many things and wonder if it could be:
- I need to set a range of ports (instead of only 1)
- I need to configurate the PCs in a specific way.

Thank you in advance

**Comment by mcgratta** [comment]
There is no limit on the number of computers that can be used for an MPI computation. However, most users running big jobs have switched over to using linux or subscribing to a cloud computing service. At NIST, we use a linux cluster in which we run jobs that use hundreds of cores over dozens of "nodes" in the cluster. We have run small jobs across our Windows network, but I would defer to others who have run bigger jobs on multiple Windows computers.
